### PR TITLE
[DCOS-54802] Add fixes for proper RLIMIT_INFINITY support.

### PR DESCRIPTION
**Pull request by kaiwalyajoshi**
Fix previous bugs with setting `RLIMIT_INFINITY` which was incorrectly set.

JIRA: DCOS-54802

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/RLimitSpec.java [hunk 1]
```diff
@@ -79,6 +85,16 @@ private void validate() throws InvalidRLimitException {
     if (soft != null && soft > hard) {
       throw new InvalidRLimitException("soft rlimit must be less than or equal to the hard rlimit");
     }
+
+    if ((soft != null && soft < RLIMIT_INFINITY) || (hard != null && hard < RLIMIT_INFINITY)) {
+      throw new InvalidRLimitException("soft and hard rlimits must be positive with the exception"
+          + " of " + RLIMIT_INFINITY + " which represents unlimited.");
+    }
+
+    if ((soft != null && soft == RLIMIT_INFINITY) ^ (hard != null && hard == RLIMIT_INFINITY)) {
```
Comment (mpereira) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/RLimitSpec.java]: Didn't know about `^` as an XOR, nice!

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java [hunk 2]
```diff
@@ -673,7 +673,13 @@ private static String getConfigTemplateDownloadPath(ConfigFileSpec config) {
           .setType(rLimit.getEnum());
 
       // RLimit itself validates that both or neither of these are present.
-      if (soft.isPresent() && hard.isPresent()) {
+      if ((soft.isPresent() && hard.isPresent()) && (soft.get() != RLimitSpec.RLIMIT_INFINITY &&
+          hard.get() != RLimitSpec.RLIMIT_INFINITY))
```
Comment (mpereira) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java]: Does it make sense to add a unit test that exercises this condition as well?

Comment (mpereira) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java]: Also, should we log something in the case where only one of `soft` or `hard` are set to `RLIMIT_INFINITY`?

Comment (kaiwalyajoshi) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java]: We currently throw an exception in the event that only one is exclusively set: 

https://github.com/mesosphere/dcos-commons/blob/709a827ee07c0aaf01cfc54c3615196ff40aaf30/sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/RLimitSpec.java#L95

Comment (kaiwalyajoshi) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java]: There's this existing test case which excercises setting normal RLimits:
https://github.com/mesosphere/dcos-commons/blob/709a827ee07c0aaf01cfc54c3615196ff40aaf30/sdk/scheduler/src/test/java/com/mesosphere/sdk/specification/RLimitSpecTest.java#L8

**Pull request merged**